### DISCUSSION TITLE
Best way to grab screens from multiple cams

### DISCUSSION THREAD

**Discussion by ibanez78** [mrlt8/docker-wyze-bridge#1292] [General]
Hello, I have been using an apple automator flow to grab screens from each of my wyze cams using wyzebridge and commands like (http://###.###.###.###:5001/snapshot/cameraname.jpg). I would say this works about 75% of the time, the other 25% percent the pictures it grabs are from earlier in the day, almost like a stuck image. Is there a better way to get screens or "force" the camera's to refresh before the pics are taken?

**Comment by mrlt8** [comment]
The snapshot endpoint will fallback to the wyze thumbnail if it fails to grab a new snapshot from the rtsp stream, so that's probably what is happening. 

How often do you need the snapshots? You could either, 1) start the stream before attempting to grab the snapshot by using the api endpoint `http://###.###.###.###:5001/api/cameraname/state/start`, or 2) set `ON_DEMAND=False` so that the bridge will keep the stream alive if you are taking multiple snapshots.

**Reply by ibanez78** [reply]
I'm taking screenshots every 30 mins

**Reply by mrlt8** [reply]
You could try adding the SNAPSHOT option to your config:
```yaml
SNAPSHOT=RTSP1800
```

or as mentioned before, you could try to start the connection to the camera before taking the screenshot:
http://###.###.###.###:5001/api/cameraname/state/start
http://###.###.###.###:5001/snapshot/cameraname.jpg
http://###.###.###.###:5001/api/cameraname/state/stop

**Comment by ibanez78** [comment]
would I add that config line to the compose file?

**Reply by mrlt8** [reply]
yes, like:
```yaml
        environment:
            - WYZE_EMAIL=...
            - WYZE_PASSWORD= ...
            - API_ID=...
            - API_KEY=...
            - SNAPSHOT=RTSP1800
```

**Reply by ibanez78** [reply]
thanks, so I tried to set them up with timestamps but it's not working, here is part of the config
            - SNAPSHOT=RTSP300
            - SNAPSHOT_FORMAT={driveway}/%Y-%m-%d/%H-%M-%S.jpg
            - SNAPSHOT_FORMAT={window}/%Y-%m-%d/%H-%M-%S.jpg

the files are just keeping the regular filenames like driveway.jpg, window.jpg
